src/audio/dai.c

```diff
@@ -65,6 +65,7 @@ struct dai_data {
 	struct dma *dma;
 	uint32_t frame_bytes;
 	bool xrun;		/* true if we are doing xrun recovery */
+	bool started;		/* true if stream was just started */
```

Comment:
What is the definition of started ? This looks like we are duplicating state information ?

Reply:
@lgirdwood Component is started, but it hasn't yet copied any data. It's really just used to avoid reporting an unecessary xrun. Couldn't come up with any another name and adding additional comp state was too much.

Reply:
This just seems like a shortcut by adding a new state indirectly and cheaply and will ultimately lead to confusion for integrators. I suggest we have STARTED and RUNNING states for components and pipeline.

Reply:
Right now this is something internal for DAI, so I don't see integrators looking into it, since it's generic component. Adding additional state will add even more confusion, because it will be used only in this one place and I will need to set it pretty much in every component's copy.

Reply:
@tlauda pipeline and component states should all be in the one place period (especially since this is a generic state that should be useful for others too), we have one user of this new state today but that does not mean there wont be other users in the future.